src/test/java/org/alfresco/AllDBTestsTestSuite.java

```diff
@@ -76,6 +75,9 @@
 
 	// From MiscContextTestSuite
     org.alfresco.repo.domain.query.CannedQueryDAOTest.class,
+
+    // REPO-2963 : Tests causing a cascade of failures in AllDBTestsTestSuite on PostgreSQL/MySQL	
+    // Moved at the bottom of the suite because DbNodeServiceImplTest.testNodeCleanupRegistry() takes a long time on a clean DB.
```

Comment:
The locks will be trying to insert rows at the same time, which is probably where the retrying is coming from.
Once the data has been populated, there will not be new inserts for the unique lock tokens.
Well done for finding this.